src/test/kotlin/service/user/DeleteUserTest.kt

```diff
@@ -0,0 +1,84 @@
+package se.uulm.snowballr.backend.service.user
+
+import io.mockk.coEvery
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.db.dummyUserId
+import se.uulm.snowballr.backend.model.SnowballRException
+import se.uulm.snowballr.backend.model.SnowballRException.UnauthorizedException
+import se.uulm.snowballr.backend.service.MainServiceTest
+import se.uulm.snowballr.backend.testCoroutine
+import snowballr.Base
+import snowballr.UserOuterClass.UserRole
+
+@ExperimentalCoroutinesApi
+@DelicateCoroutinesApi
+internal class DeleteUserTest : MainServiceTest() {
+    @Test
+    fun `When an user that is not an admin tries to delete another user, then an unauthorized exception is thrown`() =
+        testCoroutine {
+            val firstUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_DEFAULT)
+            val request =
+                Base.Id
+                    .newBuilder()
+                    .setId(firstUser.id.toString())
+                    .build()
+            val secondUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_DEFAULT)
+
+            coEvery { userRepoMock.getUserById(dummyUserId!!) } returns secondUser
```

Comment:
Here we should also mock that if the first user is requested that the first user is returned.
```suggestion
            coEvery { userRepoMock.getUserById(dummyUserId!!) } returns secondUser
            coEvery { userRepoMock.getUserById(request.id) } returns firstUser
```
The object has only null values everywhere when mocking this, and therefore the only check is ignored. Maybe we should think about, whether calls should be allowed or not if they are not mocked.

Reply:
Adapt the other test cases too to mock both of this calls.